packages/adapter/addon/json-api.js

```diff
@@ -221,6 +221,20 @@ const JSONAPIAdapter = RESTAdapter.extend({
   */
   coalesceFindRequests: false,
 
+  buildQuery(snapshot) {
+    let query = this._super(...arguments);
+
+    if (snapshot) {
+      let { fields } = snapshot;
+
+      if (fields) {
+        query.fields = fields;
+      }
+    }
+
+    return query;
+  },
```

Comment:
TODO - move to adapterOptions and document.

Reply:
TODO - track fields in shouldReloadRecord,etc and return true/false based on if different than past requests